smartsim/_core/_cli/scripts/dragon_install.py

```diff
@@ -152,40 +267,56 @@ def retrieve_asset_info() -> GitReleaseAsset:
     return asset
 
 
-def retrieve_asset(working_dir: pathlib.Path, asset: GitReleaseAsset) -> pathlib.Path:
+def retrieve_asset(
+    request: DragonInstallRequest, asset: GitReleaseAsset
+) -> pathlib.Path:
     """Retrieve the physical file associated to a given GitHub release asset
 
     :param working_dir: location in file system where assets should be written
     :param asset: GitHub release asset to retrieve
     :returns: path to the directory containing the extracted release asset"""
-    download_dir = working_dir / str(asset.id)
+    download_dir = request.working_dir / str(asset.id)
 
     # if we've previously downloaded the release and still have
     # wheels laying around, use that cached version instead
-    if download_dir.exists() or list(download_dir.rglob("*.whl")):
-        return download_dir
+    cleanup(download_dir)
 
     download_dir.mkdir(parents=True, exist_ok=True)
 
     # grab a copy of the complete asset
     asset_path = download_dir / str(asset.name)
-    download_url = asset.browser_download_url
+
+    # use the asset URL instead of the browser_download_url to enable
+    # using auth for private repositories
+    headers: t.Dict[str, str] = {"Accept": "application/octet-stream"}
+
+    if request.raw_token:
+        headers["Authorization"] = f"Bearer {request.raw_token}"
 
     try:
-        urlretrieve(download_url, str(asset_path))
-        logger.debug(f"Retrieved asset {asset.name} from {download_url}")
+        # a github asset endpoint causes a redirect. the first request
+        # receives a pre-signed URL to the asset to pass on to WebTGZ
+        dl_request = Request(asset.url, headers=headers)
+        response = urlopen(dl_request)
+        presigned_url = response.url
+
+        logger.debug(f"Retrieved asset {asset.name} metadata from {asset.url}")
     except Exception:
-        logger.exception(f"Unable to download asset from: {download_url}")
+        logger.exception(f"Unable to download {asset.name} from: {asset.url}")
+        presigned_url = asset.url
 
     # extract the asset
-    archive = WebTGZ(download_url)
-    archive.extract(download_dir)
+    try:
+        archive = _WebTGZ(presigned_url, headers=headers)
+        archive.extract(asset_path)
+        logger.debug(f"Extracted {asset.name} to {download_dir}")
+    except Exception as ex:
+        logger.exception("oops")
 
-    logger.debug(f"Extracted {download_url} to {download_dir}")
     return download_dir
 
 
-def install_package(asset_dir: pathlib.Path) -> int:
+def install_package(request: DragonInstallRequest, asset_dir: pathlib.Path) -> int:
```

Comment:
another "fix up the docstring" request here as well :)

Reply:
got it!